From 9709b2da2beb500c61be212252cd7b3adebd9b8a Mon Sep 17 00:00:00 2001
From: Etan Cohen <etancohen@google.com>
Date: Fri, 29 Jul 2016 17:25:23 -0700
Subject: [PATCH] [mDNS] Support interface names larger than 8 characters

Interface names >8 character would cause the IPv6 interface
enumeration for mDNS to fail. None of those interfaces would
get mDNS packets.

(cherry-pick of db3e830f3f3dbc8e4d545785db4ed01e3e1cc382)

Bug: 30101764
Change-Id: I761422e53d787c7e50866c1bc87eb5d2a4e95b0e
---
 mDNSPosix/mDNSUNP.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/mDNSPosix/mDNSUNP.c b/mDNSPosix/mDNSUNP.c
index d20f214..5cabbc6 100755
--- a/mDNSPosix/mDNSUNP.c
+++ b/mDNSPosix/mDNSUNP.c
@@ -86,7 +86,7 @@ struct ifi_info *get_ifi_info_linuxv6(int family, int doaliases)
     FILE *fp = NULL;
     char addr[8][5];
     int flags, myflags, index, plen, scope;
-    char ifname[9], lastname[IFNAMSIZ];
+    char ifname[IFNAMSIZ], lastname[IFNAMSIZ];
     char addr6[32+7+1]; /* don't forget the seven ':' */
     struct addrinfo hints, *res0;
     int err;
@@ -104,7 +104,7 @@ struct ifi_info *get_ifi_info_linuxv6(int family, int doaliases)
             goto gotError;
         }
         while (fscanf(fp,
-                      "%4s%4s%4s%4s%4s%4s%4s%4s %02x %02x %02x %02x %8s\n",
+                      "%4s%4s%4s%4s%4s%4s%4s%4s %02x %02x %02x %02x %15s\n",
                       addr[0],addr[1],addr[2],addr[3],
                       addr[4],addr[5],addr[6],addr[7],
                       &index, &plen, &scope, &flags, ifname) != EOF) {
-- 
2.19.2

